spec/lib/rbac/filterer_spec.rb

```diff
@@ -1286,56 +1287,143 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
         end
       end
 
-      it "returns users from current user's groups" do
-        other_user.miq_groups << group
-        get_rbac_results_for_and_expect_objects(User, [user, other_user])
-      end
+      context "with other users in the group" do
+        before do
+          other_user.miq_groups << group
+        end
+
+        it "returns users from current user's groups" do
+          get_rbac_results_for_and_expect_objects(User, [user, other_user])
+        end
 
-      it "returns user's groups" do
-        _expected_groups = [group, other_group] # this will create more groups than 2
-        get_rbac_results_for_and_expect_objects(MiqGroup, user.miq_groups)
+        it "returns user's groups" do
+          get_rbac_results_for_and_expect_objects(MiqGroup, [group])
+        end
       end
 
-      context "with self-service user" do
+      context "where the user is in another group that is a regular group" do
+        let(:other_tenant) { group.tenant } # ensure that the other group is in the same tenant
+
         before do
-          allow_any_instance_of(MiqGroup).to receive_messages(:self_service? => true)
+          other_user # ensure we create another user in the other group
+
+          user.miq_groups << other_group
         end
 
-        it "returns only the current user" do
-          get_rbac_results_for_and_expect_objects(User, [user])
+        context "and the current_group is the original regular group" do
```

Comment:
```suggestion
        context "and the current_group is a regular group" do
```

unsure the difference between this context and the next context.
Is it simply that there are or are not other users in the group?

Maybe just create a user in the group and not in the group and you'll handle the happy and sad paths?

Reply:
aah. If you use shared_group, I think the language will end up better. (so you could use shared group for a context where the group is shared)

Reply:
Yeah i used shared group